### ISSUE TITLE
What should we do with multiple expconfigs found in db by `Experiment.__init__`?

### ISSUE THREAD

**Issue by tsirif** [mila-udem/orion#57]
Case: During initialization of an `Experiment` object, we query the database for documents (configurations) with a specific (exp_name, user_name) tuple. Recently in #55 we proposed to make this tuple a key for the `experiments` collection in database. This means that we should also do probably with the checks in [lines 125-130](https://github.com/mila-udem/metaopt/pull/55/files#diff-3c00675cc564c433d5ac202df8d38c66R122) in `metaopt.core.worker.experiment`.

```python
if len(config) > 1:
    log.warning("Many (%s) experiments for (%s, %s) are available but "
                       "only the most recent one can be accessed. "
                       "Experiment forks will be supported soon.", len(config), name, user)
config = sorted(config, key=lambda x: x['metadata']['datetime'],
reverse=True)[0]
```

Now it is sure that a single document will be returned from this query, under normal conditions. Should we remove the check? Or should we convert it to error and throw an exception?? Are there other choices, what do you think?

**Comment by bouthilx** [comment]
I think we can safely remove the check. The unittests should take care of testing whether uniqueness is correctly forced upon keys when requested. Afterwards, we can assume `len(config) == 1`.

**Comment by tsirif** [comment]
So with that we assume that users are going to interact with the database
only through our software. Is this a correct assumption?

**Comment by bouthilx** [comment]
No, once the index is created it resides inside the database, hence any write on the DB would raise duplicate key errors. The only problem that could occur is if they first pre-populate the database with experiments, then the operation `ensure_index()` could fail if the database already contain duplicate keys. I'd say fiddling with the DB directly is quite likely, but pre-propulating is not since it would require them to reimplement all our structure for experiments.

Nevertheless, adding a unittest for cases where `ensure_index(..., unique=True)` is applied on a database which already contains duplicate keys would be a good idea. It would force us to find a way to cope with that.

In short, if someone ever add duplicate keys before ever running metaopt on the db, then the code would fail **before** we get to the line `len(config) > 1`, so we can remove this part.

**Comment by tsirif** [comment]
@bouthilx Ok let's remove that code!!
